kratos/input_output/vtk_output.cpp

```diff
@@ -408,18 +417,27 @@ void VtkOutput::WriteNodalResultsToFile(const ModelPart& rModelPart, std::ofstre
 void VtkOutput::WriteElementResultsToFile(const ModelPart& rModelPart, std::ofstream& rFileStream)
 {
     const auto& r_local_mesh = rModelPart.GetCommunicator().LocalMesh();
-    Parameters element_results = mOutputSettings["element_data_value_variables"];
+    Parameters element_data_value_variables = mOutputSettings["element_data_value_variables"];
+    Parameters element_flags = mOutputSettings["element_flags"];
 
     int num_elements = static_cast<int>(r_local_mesh.NumberOfElements());
     rModelPart.GetCommunicator().SumAll(num_elements);
 
     if (num_elements > 0) {
         // write cells header
         rFileStream << "CELL_DATA " << r_local_mesh.NumberOfElements() << "\n";
-        rFileStream << "FIELD FieldData " << element_results.size() << "\n";
-        for (IndexType entry = 0; entry < element_results.size(); ++entry) {
-            const std::string element_result_name = element_results[entry].GetString();
-            WriteGeometricalContainerResults(element_result_name,r_local_mesh.Elements(),rFileStream);
+        rFileStream << "FIELD FieldData " << element_data_value_variables.size() + element_flags.size() << "\n";
+        for (IndexType entry = 0; entry < element_data_value_variables.size(); ++entry) {
+            const std::string& r_element_result_name = element_data_value_variables[entry].GetString();
+            WriteGeometricalContainerResults(r_element_result_name,r_local_mesh.Elements(),rFileStream);
+        }
+
+        // Writing element_flags
+        for (IndexType entry = 0; entry < element_flags.size(); ++entry) {
+            // write nodal results variable header
+            const std::string& r_element_result_name = element_flags[entry].GetString();
+            const Flags flag = KratosComponents<Flags>::Get(r_element_result_name);
+            WriteFlagContainerVariable(rModelPart.Elements(), flag, r_element_result_name, rFileStream);
```

Comment:
I am pretty convinced that those should be the local elements and not all of them

Reply:
Done